### PR TITLE
enh(API): Interfaces for logout session use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR add the interfaces structures and test for the use case logout session.

**Fixes** # MON-11925

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    public $token;
+
+    /**
+     * @param string $token
+     */
+    public function __construct(string $token)
+    {
+        $this->token = $token;
+    }
```
Comment (adr-mo) [inline src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php:39]:
```suggestion

    /**
     * @param string $token
     */
    public function __construct(public string $token)
    {
    }
```

Comment (adr-mo) [reply src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php:39]: Lets use as much as possible PHP 8 features 💯

File: src/Core/Infrastructure/Security/Api/LogoutSession/LogoutSessionController.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        if (!isset($token)) {
+            return $this->view([
+                "code" => Response::HTTP_UNAUTHORIZED,
+                "message" => _(self::INVALID_CREDENTIALS_MESSAGE)
+            ], Response::HTTP_UNAUTHORIZED);
+        }
+        $request = new LogoutSessionRequest($token);
+        $useCase($request);
+
+        return $this->view(['message' => 'Successful logout']);
```
Comment (adr-mo) [inline src/Core/Infrastructure/Security/Api/LogoutSession/LogoutSessionController.php:55]: The error message is translated. This one could also be translated I guess

**Review by adr-mo — CHANGES_REQUESTED** [review]

**Review by adr-mo — APPROVED** [review]

File: src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+class LogoutSessionRequest
+{
+    /**
+     * @param string $token
+     */
+    public function __construct(public string $token)
+    {
+        $this->token = $token;
+    }
```
Comment (adr-mo) [inline src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php:34]:
```suggestion
    {
    }
```

Comment (adr-mo) [reply src/Core/Application/Security/UseCase/LogoutSession/LogoutSessionRequest.php:34]: This is not needed anymore

**Review by kduret — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**